src/Chainweb/Mempool/InMem.hs

```diff
@@ -454,21 +455,22 @@ recordRecentTransactions maxNumRecent newTxs rlog = rlog'
     numNewItems = V.length newTxs
     oldNext = _rlNext rlog
     newNext = oldNext + fromIntegral numNewItems
-    newTxs' = reverse ([oldNext..] `zip` V.toList newTxs)
-    newL' = newTxs' ++ _rlRecent rlog
-    newL = force $ take maxNumRecent newL'
+    newTxs' = V.reverse (V.map (T2 oldNext) newTxs)
+    newL' = newTxs' <> _rlRecent rlog
+    newL = force $ V.take maxNumRecent newL'
```

Comment:
This change worsens asymptotics here but I don't suppose we care too much, the size of the vector is bounded

Reply:
yes, I am aware of it. There are a few `memcpy` going on here. However, for and array of 2048 pointers, it may still be faster than traversing the list. It also depends on how much inlining and fusion happens here.

The proper solution would be to use a mutable fixed size ring buffer.